substrate/frame/collective/src/lib.rs

```diff
@@ -931,22 +956,17 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	}
 
 	/// Removes a proposal from the pallet, and deposit the `Disapproved` event.
-	pub fn do_disapprove_proposal(proposal_hash: T::Hash) -> u32 {
+	pub fn do_disapprove_proposal(proposal_hash: PreimageHash) -> u32 {
 		// disapproved
 		Self::deposit_event(Event::Disapproved { proposal_hash });
 		Self::remove_proposal(proposal_hash)
 	}
 
-	// Removes a proposal from the pallet, cleaning up votes and the vector of proposals.
-	fn remove_proposal(proposal_hash: T::Hash) -> u32 {
+	// Removes a proposal from the pallet, cleaning up votes
+	fn remove_proposal(proposal_hash: PreimageHash) -> u32 {
 		// remove proposal and vote
-		ProposalOf::<T, I>::remove(&proposal_hash);
 		Voting::<T, I>::remove(&proposal_hash);
-		let num_proposals = Proposals::<T, I>::mutate(|proposals| {
-			proposals.retain(|h| h != &proposal_hash);
-			proposals.len() + 1 // calculate weight based on original length
-		});
-		num_proposals as u32
+		Voting::<T, I>::count() + 1
```

Comment:
```suggestion
		Voting::<T, I>::count() + 1
```
should use safe math